lib/completion.ts

```diff
@@ -349,7 +349,15 @@ export class Completion implements CompletionInstance {
     let script = this.zshShell
       ? templates.completionZshTemplate
       : templates.completionShTemplate;
-    const name = this.shim.path.basename($0);
+    let name = this.shim.path.basename($0);
+    // Santize `name` to make sure it is a valid bash function name
+    // first, check if the first character is an alphabetic character or underscore
```

Comment:
I think the completion template might be putting an underscore on the front already so this step is not needed? (Based on code inspection, not from running code.)

https://github.com/yargs/yargs/blob/0c95f9c79e1810cf9c8964fbf7d139009412f7e7/lib/completion-templates.ts#L8

Reply:
ah, good catch